swap/src/protocol/bob/event_loop.rs

```diff
@@ -193,7 +193,7 @@ impl EventLoop {
                         }
                         OutEvent::ResponseSent => {}
                         OutEvent::CommunicationError(err) => {
-                            error!("Communication error: {:#}", err)
+                            anyhow!("Communication error: {:#}", err);
```

Comment:
This doesn't actually terminate the loop, you are just constructing an error. You need to either explicitly say `return Err(anyhow!(""))` or use `bail!` which implies the `return`.

Reply:
Fixed - changed to `bail!`! For some reason I thought that `anyhow` is equivalent to `bail`, but as you pointed out it does not return. How would it ever return whit a questionmark in the end. What a noob mistake 🤦‍♂️